lib/codebreaker/game.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+require_relative 'validate_service'
+require_relative 'secret_code'
+require_relative 'version'
+require_relative 'player'
+
+module Codebreaker
+  # Main class for codebreaker game
+  class Game
+    attr_writer :secret_code
+    attr_accessor :user_code, :player, :statistic
+    attr_reader :result, :attempts, :hints, :old_hint, :difficulty_values
+
+    DIFFICULTY_VALUES = {
+      easy: { attempts: 15, hints: 2 },
+      medium: { attempts: 10, hints: 1 },
+      hell: { attempts: 5, hints: 1 }
+    }.freeze
+    def initialize
+      @secret_code = generate_s_code
+      @user_code = []
+      @statistic = []
+      @attempts = 0
+      @hints = 0
+      @old_hint = 0
+      @difficulty_values = {}
+    end
+
+    def new_game(name, difficulty)
+      @secret_code = generate_s_code
+      @player = ::Codebreaker::Player.new(name)
+      difficulty_level(difficulty)
+    end
+
+    def play(user_code, secret_code = @secret_code)
```

Comment:
`@secret_code` is secret. U have to hide it. So u can't read and write from out of class.